code/contracts/ONEWallet.sol

```diff
@@ -105,13 +109,15 @@ contract ONEWallet {
             bytes32(bytes4(indexWithNonce)),
             eotp
         );
-        bytes32 commitHash = keccak256(bytes.concat(packed));
+        bytes32 commitHash = keccak256(bytes.concat(packed)); // IH: why do you need bytes.concat again?
         _revealPreCheck(commitHash, indexWithNonce);
         _completeReveal(commitHash);
+        // IH: until here, the code is duplicate with revealTransfer(). Can you put it into some helper function?
+
         if (lastResortAddress == address(0)) {
             return false;
         }
-        _drain();
+        _drain(); // IH: the function is short, can you expand it here? and also check return code?
```

Comment:
Yes, it should return `_drain()` rather than ignoring the return value

Reply:
addressed in https://github.com/polymorpher/one-wallet/commit/d289e3e3802c664ab73d97ec5aa79631f896af0f